### PR TITLE
IGNITE-17222 HLC propagation on transaction protocol events.

### PR DESCRIPTION
**Pull request by sergeyuttsel**

### CODE REVIEW

File: modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java [hunk 1]
```diff
@@ -192,23 +201,45 @@ public void onReceived(NetworkMessage message, NetworkAddress senderAddr, @Nulla
 
                     ReplicaRequest request = (ReplicaRequest) message;
 
+                    HybridTimestamp requestTimestamp;
```
Comment (sanpwc) [inline modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java:204]: Let's hide all that semantically consistent things into corresponding private methods, including response preparation even error one. All in all I'd expect something similar too
```
    public void start() {
        clusterNetSvc.messagingService().addMessageHandler(ReplicaMessageGroup.class, new NetworkMessageHandler() {
            @Override
            public void onReceived(NetworkMessage message, NetworkAddress senderAddr, @Nullable Long correlationId) {
                if (!busyLock.enterBusy()) {
                    throw new IgniteException(new NodeStoppingException());
                }

                try {
                    assert message instanceof ReplicaRequest : IgniteStringFormatter.format("Unexpected message [message={}]", message);

                    ReplicaRequest request = (ReplicaRequest) message;

                    HybridTimestamp requestTimestamp = extractTimestamp(request);

                    Replica replica = replicas.get(request.groupId());

                    if (replica == null) {
                        prepareReplicaUnavailableErrorResponse(senderAddr, correlationId, request, requestTimestamp);
                    }

                    CompletableFuture<Object> result = replica.processRequest(request);

                    result.handle((res, ex) -> {
                        NetworkMessage msg;

                        if (ex == null) {
                            msg = prepareReplicaResponse(requestTimestamp, res);
                        } else {
                            msg = prepareReplicaErrorResponse(request, requestTimestamp, ex);
                        }

                        clusterNetSvc.messagingService().respond(senderAddr, msg, correlationId);

                        return null;
                    });
                } finally {
                    busyLock.leaveBusy();
                }
            }
        });
    }
```

File: modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaService.java [hunk 2]
```diff
@@ -99,6 +107,10 @@ private <R> CompletableFuture<R> sendToReplica(ClusterNode node, ReplicaRequest
                 assert response instanceof ReplicaResponse : IgniteStringFormatter.format("Unexpected message response [resp={}]",
                         response);
 
+                if (response instanceof TimestampAware) {
+                    clock.update(((TimestampAware) response).timestamp());
```
Comment (sanpwc) [inline modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaService.java:111]: How it's possible to get PrimaryReplicaMissException as an messagingService.invoke result?
```
                } else if (throwable instanceof PrimaryReplicaMissException) {
                    throw (PrimaryReplicaMissException) throwable;
                }
```

Comment (sergeyuttsel) [reply modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaService.java:109]: Yes, seem that it's not need.

File: modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java (hunk 1 above)
Comment (sergeyuttsel) [reply modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java:204]: Fixed.

**Pull request closed without merging**